### DISCUSSION TITLE
Newbie creating TTS in icelandic

### DISCUSSION THREAD

**Discussion by kksteini** [coqui-ai/TTS#1580] [General Q&A] [answered]
Hi there. What an awesome project!

**Short info**
I've set myself the mission of recording around 10 hours of material. So far I have one hour and 20 minutes.
Random example of a recorded sentence: 

https://user-images.githubusercontent.com/77672665/169181387-3743c5ba-09b7-4d48-8dc2-bd06cb114940.mp4

I have a generated a text set which covers ~95% of diphones at 8000 sentences (excluding rare diphones). [Paper I came across if anyone is interested](https://aclanthology.org/2020.sltu-1.44.pdf)

# Questions
* I have 8GB of VRAM to play around with but is that perhaps not enough?
* How exactly do vocoders fit in very broad terms? Am I right to assume that with sufficient data I first train a TTS, then a Vocoder and then just put them together using tts-server f.x? I'm looking at training GlowTTS and then HiFiGAN.
* I got some very mechanical robotic results from 40 minutes of audio. It didn't improve much, I think, at an hour and 20 minutes. Is it too early to say what will happen with this little of data or can early training be indicative of anything such as batch/configuration issues?

Anyway, I am excited for this undertaking and am looking forward to evenings of recording sessions and sleepless nights of watching tensorboards :grin:

**Comment by kksteini** [comment, marked as answer]
I may have been too hasty in starting a discussion. Hopefully I'm not creating too much clutter :sweat_smile: 

* 8GB of VRAM are enough to run the GlowTTS recipe but I'm considering moving the HiFiGAN training to colab if possible since I can't run it beyond batch size 14 (RTX3070 ti).
* I had been training off of the main branch but as suggested elsewhere the dev branch had some fix for GAN I believe. Whatever the case, I'm getting some nice pitch results from HiFiGAN already. I can easily fit these models together with the tts-server and they are now working as expected, I believe.
* I can only assume that making one's own TTS is **very** data hungry and will simply continue on towards my goal of 10 hours of recorded material.

**Reply by lexkoro** [reply]
I suggest you try training a VITS model which is end2end, means it already combines tts + vocoding. You should be able to get some early results with your 1 hour of data.

**Reply by kksteini** [reply]
I quickly run out of memory on VITS unless I reduce the batch size.
I tried training with an hour of data, HiFiGAN + GlowTTS with these results: 
https://user-images.githubusercontent.com/77672665/169849528-9ea86bdb-e010-46ba-b0af-52e914a811bd.mp4

I think this is promising for an hour and can only imagine there is an uptick of quality with a larger dataset

**Comment by arif334** [comment]
My one cent:

Try `UnivNet` instead of `HifiGAN`. It is much lighter and faster than `HifiGAN` yet gives similar results.

**Reply by kksteini** [reply]
Thank you for the suggestion. I will look into it :D

**Reply by kksteini** [reply]
Yeah ok, my computer can handle that without any problems. Thanks again for the suggestion

**Comment by lumpidu** [comment]
@kksteini : are you are aware of this Icelandic language program  ? https://icelandic-lt.gitlab.io/

**Reply by lumpidu** [reply]
From there you can get much more data for training your model.
